### DISCUSSION TITLE
Observe All Queries

### DISCUSSION THREAD

**Discussion by RickEyre** [TanStack/query#1778] [Q&A] [answered]
Is there currently a way to observe all queries with one observer?

**Comment by TkDodo** [comment, marked as answer]
I think you can manually create a QueryObserver: https://react-query.tanstack.com/reference/QueryObserver

**Reply by RickEyre** [reply]
To clarify, I want to observe **all** queries. I'm not quite sure how to do this..

```js
const observer = new QueryObserver(queryClient, { queryKey: '{what query key here ?}' })
 
 const unsubscribe = observer.subscribe(result => {
   console.log(result)
   unsubscribe()
 })
 ```

If I put empty string for the query key, it does not work correctly.

**Reply by TkDodo** [reply]
not sure, have you tried _not_ passing a queryKey at all?

I just took a look how the `devtools` do it, because they also observe all queries: They take the `queryCache` from `queryClient` and subscribe to the whole `queryCache`: https://github.com/tannerlinsley/react-query/blob/e3f0ad58e8f63d1c83c0fe18d4711c8c7f52b7c2/src/devtools/devtools.tsx#L392-L396

**Reply by TkDodo** [reply]
just saw this:

```
queryCache.subscribe(() => {
  queryCache.getAll().forEach(query => {
  })
})
```

**Reply by tannerlinsley** [reply]
This is how the persist plugin works and how I would recommend people listen to the cache.

**Reply by RickEyre** [reply]
Thanks @TkDodo, @tannerlinsley. I think I should be able to accomplish what I need with this. I'll post back if I encounter anymore difficulty 👍.

**Reply by chrisl8888** [reply]
@RickEyre did this end up working pretty well ? If so there any implementations you could share in the wild ? 

Or did it end up being pretty noisy? I'm thinking of doing this to post logs onto the cloud for searching and analytics.
